### PR TITLE
add concourse worker --use-houdini

### PR DESCRIPTION
**Pull request by vito**
This is primarily so I can run Concourse on my Pixelbook with the ChromeOS Linux stuff.

I updated Houdini to support creating containers with raw:// rootfses - it's kinda neat! With this I can run real pipelines, so long as nothing needs privileged containers. Resources work, and tasks work so long as they use the `registry-image` resource for fetching the image (or any other unprivileged resource).

To use this on a Pixelbook, do the following:

* Set `CONCOURSE_GARDEN_USE_HOUDINI=true` to switch to the Houdini backend.
* Set `CONCOURSE_BAGGAGECLAIM_DISABLE_USER_NAMESPACES=true` - Houdini doesn't support user namespaces (I couldn't get them to work on my Pixelbook), so this needs to be disabled so BaggageClaim doesn't do filesystem `chown`ing.
* Unset `privileged: true` - it straight up doesn't work and the container will fail to create. (If this worked I wouldn't have had to do all this.)
* Set `cap_add: [SYS_ADMIN]` so that Houdini can create bind-mounts.
* Unset `CONCOURSE_BAGGAGECLAIM_DRIVER` - ChromeOS Linux uses `btrfs`, so it'll actually nest cleanly without the need for any loopbacks, and `overlay` actually won't work.

### CODE REVIEW

File: bin/cmd/concourse/worker_nonlinux.go [hunk 1]
```diff
@@ -21,5 +21,18 @@ func (cmd WorkerCommand) lessenRequirements(prefix string, command *flags.Comman
 }
 
 func (cmd *WorkerCommand) gardenRunner(logger lager.Logger) (atc.Worker, ifrit.Runner, error) {
-	return cmd.houdiniRunner(logger, runtime.GOOS)
+	worker := cmd.Worker.Worker()
+	worker.Platform = runtime.GOOS
```
[the remark this answers is not in the document]
Comment (vito) [reply bin/cmd/concourse/worker_nonlinux.go]: Yeah that turned out kinda nice.

I don't think there's any reason to override; tags are probably preferential for anything more specific.

**Comment by vito** [comment]
@cirocosta Good catch - I'll work on a fix in Houdini and update this PR.

**Comment by vito** [comment]
Bumped to a version of Houdini that fixes the issue @cirocosta found - tested manually with the following pipeline:

```yml
resources:
- name: booklit
  type: git
  source:
    uri: https://github.com/vito/booklit

- name: 3s
  type: time
  source: {interval: 3s}

jobs:
- name: unit
  plan:
  - get: booklit
    trigger: true
  - get: 3s
  - task: test
    file: booklit/ci/test.yml

- name: demo-busybox
  plan:
  - task: demo
    config:
      platform: linux

      image_resource:
        type: registry-image
        source: {repository: busybox}

      run:
        path: find
        args: [/bin]
```

This demonstrates absolute paths working (`/opt/resource/check` etc.), relative paths working (via the booklit unit tests task), and path lookup working (via the `demo-busybox` job).

**Comment by vito** [comment]
@topherbullock @cirocosta Do either of you have time to re-review?

**Comment by cirocosta** [comment]
Hey, did the changes suggested by @topherbullock get applied? Looking at 27833f7, the assignment to `err` seems to still not be there, making it not buildable 🤔

```sh
git status
HEAD detached at 27833f7b2
nothing to commit, working tree clean

go install -v ./bin/cmd/concourse/
# github.com/mattn/go-sqlite3
sqlite3-binding.c:20515:17: warning: 'OSAtomicCompareAndSwapPtrBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_compare_exchange_strong() from <stdatomic.h> instead [-Wdeprecated-declarations]
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/libkern/OSAtomicDeprecated.h:547:6: note: 'OSAtomicCompareAndSwapPtrBarrier' has been explicitly marked deprecated here
github.com/concourse/concourse/bin/cmd/concourse
# github.com/concourse/concourse/bin/cmd/concourse
bin/cmd/concourse/worker_nonlinux.go:27:15: undefined: err
bin/cmd/concourse/worker_nonlinux.go:28:5: undefined: err
bin/cmd/concourse/worker_nonlinux.go:29:29: undefined: err
```

**Pull request merged**